elements/lisk-dpos/src/delegates_list.ts

```diff
@@ -152,6 +158,36 @@ export const shuffleDelegateListForRound = (
 	return delegateList;
 };
 
+export const shuffleDelegateListBasedOnRandomSeed = (
+	previousRoundSeed1: string,
+	previousDelegateList: ReadonlyArray<string>,
+): ReadonlyArray<string> => {
+	const delegateList: DelegateListWithRoundHash[] = [
+		...previousDelegateList,
+	].map(delegate => ({ address: delegate })) as DelegateListWithRoundHash[];
+	for (const delegate of delegateList) {
+		// tslint:disable-next-line:no-magic-numbers
+		const addressBuffer = Buffer.alloc(8);
+		addressBuffer.writeBigUInt64BE(BigInt(delegate.address.slice(0, -1)));
+		const seedSource = Buffer.concat([
+			Buffer.from(previousRoundSeed1, 'hex'),
+			addressBuffer,
+		]);
+		delegate.roundHash = hash(seedSource);
+	}
+
+	delegateList.sort((delegate1, delegate2) => {
+		const diff = delegate1.roundHash.compare(delegate2.roundHash);
+		if (diff !== 0) {
+			return diff;
+		}
+
+		return delegate1.address.localeCompare(delegate2.address);
```

Comment:
```suggestion
		return delegate1.address.localeCompare(delegate2.address, 'en');
```

Reply:
I saw in some places we are missing the localeCompare with locale `en`. maybe create an issue and address in this PR or different PR?

Reply:
I think only place it's missing is test or function that will be deleted by #5074